core/src/types.rs

```diff
@@ -613,3 +614,82 @@ impl From<Base64> for String {
 		general_purpose::STANDARD.encode(value.0)
 	}
 }
+
+#[derive(Clone, Debug)]
+pub struct ProjectName(pub String);
+
+impl ProjectName {
+	pub fn new<S: Into<String>>(name: S) -> Self {
+		ProjectName(name.into().to_case(Case::Snake))
+	}
+}
+
+impl fmt::Display for ProjectName {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "{}", self.0)
+	}
+}
+
+impl Default for ProjectName {
+	fn default() -> Self {
+		ProjectName::new("avail")
+	}
+}
+
+impl Serialize for ProjectName {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: Serializer,
+	{
+		serializer.serialize_str(&self.0)
+	}
+}
+
+impl<'de> Deserialize<'de> for ProjectName {
+	fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+	where
+		D: Deserializer<'de>,
+	{
+		let formatted_name = String::deserialize(deserializer)?.to_case(Case::Snake);
```

Comment:
Lets just document this in project_name field of the configuration, to make constraint explicit.